prog/kubernetes/provision_kubernetes_node.rb

```diff
@@ -131,4 +131,39 @@ def write_hosts_file_if_needed(ip = nil)
 
     nap 65536
   end
+
+  label def install_cni
+    script = <<BASH_SCRIPT
+#!/bin/bash
+cd /home/ubi
+exec ./kubernetes/bin/ubicni
+BASH_SCRIPT
+
+    cni_config = <<CONFIG
+{
+  "cniVersion": "1.0.0",
+  "name": "ubicni-network",
+  "type": "ubicni",
+  "ranges":{
+      "subnet_ipv6": "#{vm.ephemeral_net6.network}/80",
+      "subnet_ula_ipv6": "#{vm.nics.first.private_ipv6}",
+      "subnet_ipv4": "#{vm.nics.first.private_ipv4}"
+  }
+}
+CONFIG
+
+    sshable = vm.sshable
+    sshable.cmd("sudo tee /opt/cni/bin/ubicni", stdin: script)
+    sshable.cmd("sudo tee /etc/cni/net.d/ubicni-config.json", stdin: cni_config)
+
+    sshable.cmd("sudo chmod +x /opt/cni/bin/ubicni")
+    vm.sshable.cmd <<BASH
+set -ueo pipefail
+sudo nft add table ip nat
+sudo nft add chain ip nat POSTROUTING { type nat hook postrouting priority 100';' }
+sudo nft add rule ip nat POSTROUTING ip saddr #{vm.nics.first.private_ipv4} oifname ens3 masquerade
+BASH
```

Comment:
nit: `<<~` removes leading whitespaces. The indentation makes it more readable
https://docs.ruby-lang.org/en/3.2/syntax/literals_rdoc.html#label-Here+Document+Literals

```suggestion
    vm.sshable.cmd <<~BASH
        set -ueo pipefail
         sudo nft add table ip nat
         sudo nft add chain ip nat POSTROUTING { type nat hook postrouting priority 100';' }
         sudo nft add rule ip nat POSTROUTING ip saddr #{vm.nics.first.private_ipv4} oifname ens3 masquerade
    BASH
```